main.py

```diff
@@ -2040,9 +2048,10 @@ def show_hover(self, point, contents):
 def preserve_whitespace(contents: str) -> str:
     """Preserve empty lines and whitespace for markdown conversion."""
     contents = contents.strip(' \t\r\n')
-    contents = contents.replace('\t', '&nbsp;' * 4)
-    contents = contents.replace('  ', '&nbsp;' * 2)
-    contents = contents.replace('\n\n', '\n&nbsp;\n')
+    contents = contents.replace('\r\n', '\n')
```

Comment:
Which language-server returns `\r\n`?

Reply:
Should be all of them, schorsing to protocol?

Reply:
jsts in my case at least